src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Services/Implementation/PolicyAdministrationPoint.cs

```diff
@@ -99,6 +100,219 @@ public async Task<List<Rule>> TryWriteDelegationPolicyRules(List<Rule> rules)
+                List<Rule> currentPolicyRules = new List<Rule>();
+                foreach (XacmlRule xacmlRule in existingDelegationPolicy.Rules)
+                {
+                    currentPolicyRules.Add(PolicyHelper.CreateRuleFromPolicyAndRuleMatch(policiyToDelete, xacmlRule));
+                }
+
+                existingDelegationPolicy.Rules.Clear();
+
+                // Write delegation policy to blob storage
+                MemoryStream dataStream = PolicyHelper.GetXmlMemoryStreamFromXacmlPolicy(existingDelegationPolicy);
+                Response<BlobContentInfo> response = await _policyRepository.WritePolicyConditionallyAsync(policyPath, dataStream, leaseId);
+                if (response.GetRawResponse().Status != (int)HttpStatusCode.Created)
+                {
+                    _logger.LogError("Writing of delegation policy at path: {0} failed. Is delegation blob storage account alive and well?", policyPath, response.GetRawResponse());
+                    return null;
+                }
+
+                // Write delegation change to postgresql
+                bool postgreSuccess = await _delegationRepository.InsertDelegation($"{org}/{app}", policiyToDelete.PolicyMatch.OfferedByPartyId, coveredByPartyId, coveredByUserId, policiyToDelete.DeletedByUserId, policyPath, response.Value.VersionId, true);
+                if (!postgreSuccess)
+                {
+                    // Comment:
+                    // This means that the current version of the root blob is no longer in sync with changes in authorization postgresql delegation.delegatedpolicy table.
+                    // The root blob is in effect orphaned/ignored as the delegation policies are always to be read by version, and will be overritten by the next delegation change.
+                    _logger.LogError("Writing of delegation change to authorization postgresql database failed for changes to delegation policy at path: {0}. is authorization postgresql database alive and well?", policyPath);
+                    return null;
+                }
+
+                return currentPolicyRules;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"An exception occured while processing rules to delete in policy: {policyPath}");
+                return null;
+            }
+            finally
+            {
+                _policyRepository.ReleaseBlobLease(policyPath, leaseId);
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<Rule>> TryDeleteDelegationPolicies(List<RequestToDelete> policiesToDelete)
+        {
+            List<Rule> result = new List<Rule>();
+
+            foreach (RequestToDelete policiyToDelete in policiesToDelete)
+            {
+                List<Rule> currentRules = await DeleteRulesInPolicy(policiyToDelete);
```

Comment:
Missing try/catch handling for each call to DeleteRulesInPolicy?
In order to handle partially completed request

Reply:
It does not throw anything. The null check filter out any return without any deletion.